.github/workflows/sync-with-upstream.yaml

```diff
@@ -0,0 +1,81 @@
+  contents: read
+env:
+  # This is the username and email for the user who creates a branch and commits
+  # the changes. In an organisation that should be a dedicated devops account.
+  USER_NAME: jenstroeger
+  USER_EMAIL: jenstroeger@users.noreply.github.com
+
+jobs:
+  sync:
+    if: github.repository != 'jenstroeger/python-package-template'
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+
+    - name: Check out template repository
+      uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      with:
+        # If you decide to change the upstream template repository to a private one, uncomment
+        # the following argument to pass the required token to be able to check it out.
+        # token: ${{ secrets.<TEMPLATE_REPO_TOKEN> }}
+        repository: jenstroeger/python-package-template
+        fetch-depth: 0
+        path: template
+
+    - name: Check out current repository
+      uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      with:
+        token: ${{ secrets.REPO_ACCESS_TOKEN }}
+        fetch-depth: 0
+        ref: main
+        path: repo
+
+    - name: Sync with template
+      env:
+        GH_TOKEN: ${{ secrets.REPO_ACCESS_TOKEN }}
+      run: |
+        LATEST_VERSION=$(cd template && git describe --tags --abbrev=0)
+        CURRENT_VERSION=$(test -f repo/.github/workflows/.template_version \
+          && cat repo/.github/workflows/.template_version || echo "v0.0.0")
+        echo "Latest version is $LATEST_VERSION and current version is $CURRENT_VERSION."
+
+        if [ $CURRENT_VERSION == $LATEST_VERSION ]; then
+          echo "Unable to find a new version, exiting..."
+        else
+
+          # Copy the files from the template to the current repo.
+          rsync --recursive --verbose --verbose --exclude tests/ --exclude src/ \
+            --exclude .git/ --exclude .github/workflows/.template_version template/ repo/
+
+          # Check if the branch exists in the current repo.
+          cd repo/
+          if [ $(git rev-parse --verify origin/sync-$LATEST_VERSION 2>/dev/null) ]; then
+            echo "Branch sync-$LATEST_VERSION already exists. Exiting..."
+            exit 0
+          fi
+
+          # Create a branch, commit, and push the changeset.
+          git checkout -b sync-$LATEST_VERSION
+          echo $LATEST_VERSION > .github/workflows/.template_version
+          git add .
+          git config --global user.name $USER_NAME
+          git config --global user.email $USER_EMAIL
+          git config --list --global # For debug purposes.
+          git commit -m "chore: sync with template"
+          git push --set-upstream origin sync-$LATEST_VERSION
+
+          # Create the pull request.
+          gh pr create --base main --title "chore: sync with package template $LATEST_VERSION" \
+            --body "This PR was automatically generated." --head sync-$LATEST_VERSION
```

Comment:
Should we add the [`--draft`](https://cli.github.com/manual/gh_pr_create#options) option here?

Reply:
No because `--draft` implied that the PR is not ready for review.

Reply:
Hmph.